src/Hedgehog/Property.fs

```diff
@@ -1,11 +1,33 @@
-﻿namespace Hedgehog
+namespace Hedgehog
 
 open System
 
 [<Struct>]
 type Property<'a> =
     | Property of Gen<Journal * Outcome<'a>>
 
+
+type PropertyConfig = internal {
+    TestLimit : int<tests>
+    ShrinkLimit : int<shrinks> option
+}
+
+
+module PropertyConfig =
+    /// The default configuration for a property test.
+    let defaultConfig : PropertyConfig =
+        {   TestLimit = 100<tests>
+            ShrinkLimit = None }
+
+    /// The number of shrinks to try before giving up on shrinking.
+    let withShrinkLimit (shrinkLimit : int<shrinks>) (config : PropertyConfig) : PropertyConfig =
+        { config with ShrinkLimit = Some shrinkLimit }
```

Comment:
Note that `shrinkLimit` is _not_ an `Option`. This means that they can't set the value to `None`. This makes for a somewhat cleaner API (`PropertyConfig.withShrinkLimit 0` vs `PropertyConfig.withShrinkLimit (Some 0)`) but I don't feel strongly about it.

Reply:
Since calling this function is an explicit choice a user would make, I'm fine with the parameter being a simple `int<shrinks>`. Making the parameter an `Option<int<shrinks>>` exposes how we've chosen to implement this a bit too much. Also, if we later decided to make this required, this could be very confusing. Consider this code:

```fsharp
PC.defaultConfig |> PC.withShrinkLimit None
```

If we decided to require that parameter, it would be very surprising when a test failed due to a shrink limit being imposed when the user explicitly chose `None`.

Reply:
You could add to the API `withoutShrinkLimit`

Reply:
Good idea. I was torn between that and `withNoShrinkLimit`, but my libarts sisters talked me out of it.

Inspired by [Haskell](https://hackage.haskell.org/package/hedgehog-1.0.4/docs/Hedgehog-Internal-Property.html#v:withShrinks), I dropped "Limit" from `withShrinkLimit` and `withTestLimit`, but kept "limit" in the parameter name.